mfkdf2-web/test/mfkdf2/security.test.ts

```diff
@@ -0,0 +1,843 @@
+/* eslint no-unused-expressions: "off" */
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+chai.should();
+
+import { suite, test } from 'mocha';
+import mfkdf, { uniffiInitAsync } from '../../src/api';
+import crypto, { hkdfSync } from 'crypto';
+import { Mfkdf2Error } from '../../src/generated/web/mfkdf2';
+import speakeasy from 'speakeasy';
+
+function xor(a, b) {
+  const length = Math.max(a.length, b.length)
+  const buffer = Buffer.alloc(length)
+
+  for (let i = 0; i < length; ++i) {
+    buffer[i] = a[i] ^ b[i]
+  }
+
+  return buffer
+}
+
+
+suite('mfkdf2/security', () => {
+  before(async () => {
+    await uniffiInitAsync();
+  });
+
+  suite('factor-fungibility', () => {
+    test('correct', async () => {
+      const setup = await mfkdf.policy.setup(
+        await mfkdf.policy.and(
+          await mfkdf.setup.factors.password('password1', { id: 'password1' }),
+          await mfkdf.setup.factors.password('password2', { id: 'password2' })
+        )
+      )
+
+      const derive = await mfkdf.policy.derive(setup.policy, {
+        password1: await mfkdf.derive.factors.password('password1'),
+        password2: await mfkdf.derive.factors.password('password2')
+      })
+
+      derive.key.toString('hex').should.equal(setup.key.toString('hex'))
+    })
+
+    test('incorrect', async () => {
+      const setup = await mfkdf.policy.setup(
+        await mfkdf.policy.and(
+          await mfkdf.setup.factors.password('password1', { id: 'password1' }),
+          await mfkdf.setup.factors.password('password2', { id: 'password2' })
+        )
+      )
+
+      const derive = await mfkdf.policy.derive(
+        setup.policy,
+        {
+          password1: await mfkdf.derive.factors.password('password2'),
+          password2: await mfkdf.derive.factors.password('password1')
+        },
+        false
+      )
+
+      derive.key.toString('hex').should.not.equal(setup.key.toString('hex'))
+    })
+  })
+
+  /* TODO (@lonerapier): Implement this test after sharks repo is updated
+  suite('share-indistinguishability', () => {
```

Comment:
this test will be updated after core lib has a secrets module which is blocked on sharks right now.